### PR TITLE
Configurable "terms of use" page.

### PR DESCRIPTION
**Pull request by andorp**
Fixes: #829

### CODE REVIEW

File: tests/Frontend/CoreSpec.hs [hunk 1]
```diff
@@ -408,10 +408,11 @@ postToForm (FormTest g c check) = do
         page <- pick g
         ctx <- pick (arbFormPagePayloadCtx page)
         mpayload <- pick (arbFormPageInvalidPayload page)
-        forM_ mpayload
-            (\payload -> do
+        case mpayload of
+            Nothing -> liftIO $ pendingWith "*In*valid form input is not defined for this."
```
Comment (fisx) [inline tests/Frontend/CoreSpec.hs]: :+1:

File: src/Frontend/Page/Static.hs [hunk 2]
```diff
@@ -48,8 +52,9 @@ instance Page PageStaticTermsOfUse where
...
-        h1_ "Nutzungsbedingungen"
-        p_  "Lorem ipsum"
-        p_  "Duis aumet"
-        p_  "Ut wisi"
+    toHtml p@(PageStaticTermsOfUse terms) = semanticDiv p .
+        div_ [class_ "text-markdown"] $
+            terms ^. html
+
+termsOfUse :: ActionM m => m PageStaticTermsOfUse
+termsOfUse = PageStaticTermsOfUse <$> query Persistent.termsOfUse
```
Comment (fisx) [inline src/Frontend/Page/Static.hs]: the "terms of use" page shouldn't be in this module any more.

Comment (fisx) [inline src/Frontend/Page/Static.hs]: more general remark: this module makes a good case against structuring the page modules along the sections "template" and "page".  we should have sections for pages, and those sections should contain first the page instances and then the handler.

**Comment by fisx** [comment]
did you test this?  doesn't work for me yet:

- the initial text in the edit form is empty (instead of the admittedly boring template text we have in the static page)
- the page doesn't show the content i entered, but the static page.
- the static page html file hasn't been removed from `/static/'.

**Comment by fisx** [comment]
"initial text" issue can be postponed to #199.

**Comment by andorp** [comment]
> the page doesn't show the content i entered, but the static page.

I think it is a caching issue as I checked it and I couldn't reproduce it.

> the static page html file hasn't been removed from `/static/'

The original text was defined in a handler, there is no file to delete.

**Comment by fisx** [comment]
rename `Frontend.Page.Static` to `Frontend.Page.Info` as discussed?

**Comment by fisx** [comment]
thanks.  and sorry for the sloppy review...

**Pull request merged**